functions.ts

```diff
@@ -59,25 +59,19 @@ function printUpcomingOpenings(place: Place): string {
     var time: number = getExtendedTime();
     var output: string = "";
 
-    for (var i=0; i<hours.length; i++) {
-        var x: Interval = hours[i];
-        // Add all of the intervals that haven't ended yet to the output
-        if (time < x.getClose()) {
-            output += x.getIntervalString() + " and ";
-        }
-    }
+    // Add all of the intervals that haven't ended yet to the output
+    output += hours.filter(x => time < x.getClose()).map(x => x.getIntervalString()).join(" and ");
```

Comment:
You can just initialize `output` to this value rather than appending it to an empty string